src/components/vm/nics/nicBody.tsx

```diff
@@ -271,3 +290,311 @@ export const NetworkTypeAndSourceRow = ({
         </>
     );
 };
+
+export function portForwardText(pf: VMInterfacePortForward): string {
+    let text = "";
+
+    if (pf.address)
+        text += pf.address;
+    if (pf.dev)
+        text += pf.dev;
+    if (!pf.range.some(r => r.exclude != "yes")) {
+        if (text != "")
+            text += ", ";
+        text += _("all ports");
+    }
+    for (let i = 0; i < pf.range.length; i++) {
+        const r = pf.range[i];
+        if (text != "")
+            text += ", ";
+        if (r.exclude == "yes")
+            text += _("excluding ");
+        text += r.start;
+        if (r.end)
+            text += "-" + r.end;
+        if (r.to)
+            text += " → " + r.to + (r.end ? "-" + (Number(r.to) + Number(r.end) - Number(r.start)).toString() : "");
+        if (pf.proto && pf.proto != "tcp")
+            text += "/" + pf.proto;
```

Comment:
This added line is not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-2331-3b8315e1-20251022-060022-fedora-42-devel/Coverage/lcov/github-pr.diff.gcov.html)